src/main/java/com/connexta/store/service/api/StoreService.java

```diff
@@ -44,6 +45,10 @@ InputStream retrieveIrm(
       @Pattern(regexp = "^[0-9a-zA-Z]+$") @Size(min = 32, max = 32) final String datasetId)
       throws RetrieveException;
 
+  /** The caller is responsible for closing the returned {@link java.io.InputStream}. */
+  @NotNull
+  InputStream retrieveMetacard(@NotBlank final String datasetId) throws RetrieveException;
```

Comment:
❗️This is not the datasetId

Reply:
Updated so the metacard is ingested with the datasetId